README.md

```diff
@@ -1,2 +1,359 @@
+workflow. In order to make it a valid `template` in a separate file you should use the
+full operator specification from the last step instead of only using the name of the
+operator.
+
+```yaml
+  name: generate_sky
+  inputs:
+    parameters:
+      - name: desired_irradiance
+        value: 100000   # default value which can be overwritten
+  operator: radiance-operator
+  # commands and args will be used both locally and inside the container
+  command:
+    - /bin/sh
+    - -c
+  args:
+    - gensky -c -B '{{ inputs.parameters.desired_irradiance }}' > '{{ workflow.artifacts.sky }}'
+  outputs:
+    artifacts:
+      - name: sky
+        path: '{{ workflow.artifacts.sky }}'
+```
+
+Alternatively, you can use `honeybee-radiance` operator which has a command for
+generating sky.
+
+```yaml
+  name: generate_sky
+  inputs:
+    parameters:
+      - name: desired_irradiance
+        value: 100000   # default value which can be overwritten
+  operator: honeybee-radiance
+  command:
+    - honeybee-radiance
+  args:
+    - fixed-irradiance-sky
+    - --irradiance
+    - "{{inputs.parameters.desired_irradiance}}"
+  outputs:
+    artifacts:
+      - name: sky
+        path: '{{ workflow.artifacts.sky }}'
+```
+
+## 4. flow
+
+`flow` is the heart of the workflow to define how the templates should be executed.
+Queenbee supports two different types of workflows. DAGFlow and SteppedFlow. 
+
+It's common to use the output of one step as an input for another step or reference one
+of the workflow inputs as an input for one of the steps or tasks. Queenbee supports below
+words as prefix variable name:
+
+- workflow: "{{workflow.xx.yy}} is used for workflow level parameters.
+- steps: "{{steps.step_name.xx.yy}} is used in chained templates.
+- tasks: "{{tasks.task_name.xx.yy}} is used in DAG task to refer to other tasks.
+- inputs: "{{inputs.xx.yy}}" is used in operators.
+- item: "{{item}}" or "{{item.key_name}}" is used in loops. You can change item to a
+  different keyword by setting up `loop_var` in `loop_control`.
+
+Here is a example of creating the sky and generating and octree as two consecutive steps.
+Note that the values are place-holders and can be overwritten by input parameters file.
+
+```yaml
+
+flow:
+  - steps:
+    - name: generate_sky_step
+      template: generate_sky
+      inputs:
+        parameters:
+          - name: desired_irradiance
+            value: 100000
+    - name: generate_octree_step
+      template: generate_octree
+      inputs:
+        parameters:
+           - name: scene_files
+             value:
+               - "{{steps.generate_sky_steps.outputs.sky_file}}"
+               - "{{workflow.inputs.parameters.folder}}"/model/static/scene.mat
+               - "{{workflow.inputs.parameters.folder}}"/model/static/scene.rad
+
+```
+
+We can add generating sensors and running the ray-tracing itself to the current flow but
+there is a difference between these new two steps and what we had before. In previous
```

Comment:
The grammar of these two sentences is making it tough for me to understand what you mean to say but I think you mean something like this:
`We can generate sensors and run the ray-tracing using the current flow but
there is a difference between these two new steps and what we had before`.
I have to be honest that I don't know what `we had before`.  Do you mean the example above?

Reply:
we had before = the first two steps that we had as our steps before.

Reply:
What about this:

Now let's think about a longer workflow which also includes ray-tracing using the
generated octree. We need to add two new steps to the workflow:

1. generate sensor grids
2. run ray-tracing

But there is a difference between these new two steps and the initial two steps.
In the first two steps the second step of generating the octree was dependant on the
first step and it couldn't be executed until generating sky is finished. However for
generating the sensor grids we we do not need to wait for generating sky to be finished.
Finally, the last step of ray-tracing will need both the grid and the octree.